src/components/StakingVaults/AllEarnOpportunities.tsx

```diff
@@ -21,27 +26,60 @@ export const AllEarnOpportunities = () => {
   const history = useHistory()
   const location = useLocation()
   const foxyInvestorFeatureFlag = useAppSelector(state => selectFeatureFlag(state, 'FoxyInvestor'))
+  const cosmosInvestorFlag = useAppSelector(state => selectFeatureFlag(state, 'CosmosInvestor'))
   const {
     state: { isConnected },
     dispatch,
   } = useWallet()
   const sortedVaults = useSortedYearnVaults()
   const { opportunities } = useFoxyBalances()
+  const { activeStakingOpportunities, stakingOpportunities } = useCosmosStakingBalances({
+    assetId: 'cosmos:cosmoshub-4/slip44:118',
+  })
+
+  const { cosmosGetStarted, cosmosStaking } = useModal()
 
   const foxyRows = foxyInvestorFeatureFlag ? opportunities : []
+  const cosmosActiveStakingArray = cosmosInvestorFlag ? activeStakingOpportunities : []
+  const cosmosStakingArray = cosmosInvestorFlag ? stakingOpportunities : []
 
   const allRows = useNormalizeOpportunities({
     vaultArray: sortedVaults,
     foxyArray: foxyRows,
+    cosmosActiveStakingArray,
+    cosmosStakingArray,
   })
 
   const handleClick = useCallback(
     (opportunity: EarnOpportunityType) => {
-      const { type, provider, contractAddress, chain, tokenAddress, rewardAddress } = opportunity
+      const {
+        type,
+        provider,
+        contractAddress,
+        chain,
+        tokenAddress,
+        rewardAddress,
+        assetId,
+        cryptoAmount,
+      } = opportunity
       if (!isConnected) {
         dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
         return
       }
+
+      if (type === DefiType.TokenStaking && chain === ChainTypes.Cosmos) {
+        if (bnOrZero(cryptoAmount).gt(0)) {
+          cosmosStaking.open({
+            assetId,
+            validatorAddress: contractAddress,
```

Comment:
Not really a fan of keeping this as `contractAddress` but that's fine by me until we find a way to still be protocol/chain-agnostic and represent an opportunity address (whether it is a contract in e.g EVM chains or a validator in PoS chains with protocol-level staking)

Reply:
Yes I was reluctant to modify this variable name as it is already used in other opportunities. Another option would have been to create a separate new variable but we already have a lot of variables in opportunities.